packages/nimble-components/src/table-column/date-text/index.ts

```diff
@@ -82,7 +83,7 @@ export class TableColumnDateText extends TableColumnTextBase {
     @attr({ attribute: 'custom-format-matcher' })
     public customFormatMatcher: FormatMatcherAlgorithm;
 
-    @attr({ attribute: 'custom-hour12', mode: 'boolean' })
+    @attr({ attribute: 'custom-hour12', converter: optionalBooleanConverter })
```

Comment:
@rajsite, this is being done to support three different functionalities for this property, where true = 12 hour, false = 24 hour, and undefined = locale dependent. Would it be more conventional to represent this API as an enum for Nimble, and then just translate it appropriately under the hood, or do we feel we should align our API directly to that of `Intl.NumberFormat`?

Reply:
I like the suggestion of creating an enum in nimble for this. I think a nullable boolean will likely cause problems (or at the least confusion) in our API since it does diverge from what is standardly done.

Reply:
I'm not sure about the new enum. Having the values actually be `true`, `false`, and `undefined` within the enum and still requiring a custom converter doesn't seem correct to me. I think if we're going to make an enum for this, it should be an enum following our existing conventions where the enum values map to a string (or `undefined`), and then there isn't a custom converter that is required.

Reply:
+1.

Reply:
Changed enum to have string values and made resulting updates.